Source/ARTDeviceDetails.m

```diff
@@ -0,0 +1,50 @@
+//
+//  ARTDeviceDetails.m
+//  Ably
+//
+//  Created by Ricardo Pereira on 07/02/2017.
+//  Copyright © 2017 Ably. All rights reserved.
+//
+
+#import "ARTDeviceDetails.h"
+#import "ARTDevicePushDetails.h"
+
+NSString *const ARTDevicePlatform = @"ios";
+
+NSString *ARTDeviceFormFactorToStr(ARTDeviceFormFactor formFactor) {
+    switch (formFactor) {
+        case ARTDeviceFormFactorMobile:
+            return @"mobile"; //0
+        case ARTDeviceFormFactorTablet:
+            return @"tablet"; //1
+        case ARTDeviceFormFactorDesktop:
+            return @"desktop"; //2
+        case ARTDeviceFormFactorEmbedded:
+            return @"embedded"; //3
+    }
+}
+
+@implementation ARTDeviceDetails
+
++ (instancetype)fromLocalDevice {
+    return [[ARTDeviceDetails alloc] init];
```

Comment:
@tcard I forgot about it, 👍 . So I should initialise it with the `deviceId` that is stored and the current `updateToken`, if it's in memory?